core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -570,162 +791,107 @@ private BlockStoreLocation commitBlockInternal(long sessionId, long blockId, boo
-        long blockId = entry.getBlockId();
-        BlockStoreLocation oldLocation = entry.getSrcLocation();
-        BlockStoreLocation newLocation = entry.getDstLocation();
-        MoveBlockResult moveResult;
+  private void freeSpaceInternal(long sessionId, long minAvailableBytes, long maxAvailableBytes,
+      BlockStoreLocation location) throws WorkerOutOfSpaceException, IOException {
+    // TODO(ggezer): TV2 - Too much memory pressure when pinned-inodes list is large.
+    BlockMetadataEvictorView evictorView = getUpdatedView();
+    long minBytesToFree = minAvailableBytes - evictorView.getAvailableBytes(location);
+    long maxBytesToFree = maxAvailableBytes - evictorView.getAvailableBytes(location);
+
+    int blocksIterated = 0;
+    int blocksRemoved = 0;
+    int spaceFreed = 0;
+
+    LOG.debug(
+        "Failed to free space. Space requested: {}-{}, Blocks iterated: {}, Blocks removed: {},"
+            + " Freed Space: {}, Remaining: {}",
+        minAvailableBytes, maxAvailableBytes, blocksIterated, blocksRemoved, spaceFreed,
+        maxBytesToFree);
+
+    Iterator<Long> evictionCandidates =
+        mBlockIterator.getIterator(location, BlockOrder.Natural);
+    while (evictionCandidates.hasNext() && maxBytesToFree > 0) {
+      long blockToDelete = evictionCandidates.next();
+      blocksIterated++;
+      if (evictorView.isBlockEvictable(blockToDelete)) {
         try {
-          moveResult = moveBlockInternal(Sessions.createInternalSessionId(),
-              blockId, oldLocation, newLocation);
-        } catch (InvalidWorkerStateException e) {
-          // Evictor is not working properly
-          LOG.error("Failed to demote blockId {}, this is temp block", blockId);
-          continue;
-        } catch (BlockAlreadyExistsException e) {
-          continue;
-        } catch (BlockDoesNotExistException e) {
-          LOG.info("Failed to demote blockId {}, it could be already deleted", blockId);
-          continue;
-        }
-        if (moveResult.getSuccess()) {
+          BlockMeta blockMeta = mMetaManager.getBlockMeta(blockToDelete);
+          removeBlockInternal(blockMeta);
+          blocksRemoved++;
           synchronized (mBlockStoreEventListeners) {
             for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
-              listener.onMoveBlockByWorker(sessionId, blockId, moveResult.getSrcLocation(),
-                  newLocation);
+              listener.onRemoveBlockByClient(sessionId, blockMeta.getBlockId());
+              listener.onRemoveBlock(sessionId, blockMeta.getBlockId(),
+                  blockMeta.getBlockLocation());
             }
           }
+          minBytesToFree -= blockMeta.getBlockSize();
+          maxBytesToFree -= blockMeta.getBlockSize();
+          spaceFreed += blockMeta.getBlockSize();
+        } catch (BlockDoesNotExistException e) {
+          LOG.info("Failed to evict blockId {}, it could be already deleted", blockToDelete);
```

Comment:
Should this be warn?

Also, it is always the case that the block doesn't exist right? The message sounds like the eviction failed, and one possible reason is that the block is already gone.

Reply:
Yes.
It's printed out when we want to remove a block, based on annotations, and that block was not there. It can happen if block annotator is out of sync with the store.